### DISCUSSION TITLE
TinTin++ 2.02.10 #history doesn't seem to support variables

### DISCUSSION THREAD

**Discussion by jasonhsw** [scandum/tintin#44] [Bugs]
I have scripts which load different files based on the name of the session. One of the files that gets loaded is a command history file.
I have an _session variable which holds the name of the session, and the path where its files are stored, at $_session[name] and $_session[dir].
However, in V2.02.10, it seems like variable names given to the #history read and #history write commands aren't being properly substituted. I get errors about not being able to open {$_session[dir]/$_session[name].tch} for reading/writing. Even if I try loading it with the variable names in braces, it still doesn't work.

**Comment by scandum** [comment]
Thanks, it's been fixed for the next release. You can update to the latest beta at https://mudhalla.net/tintin-beta.tar.gz or you can work around it using:
```
#line sub var #history read {$_session[dir]/$_session[name].tch}
```
